backend/core/interactem/core/models/canonical.py

```diff
@@ -48,9 +48,12 @@ class CanonicalEdge(BaseModel):
     output_id: CanonicalPortID | CanonicalOperatorID
 
 
-class CanonicalPipeline(BaseModel):
-    id: CanonicalPipelineID
-    revision_id: int
+class CanonicalPipelineData(BaseModel):
     operators: Sequence[CanonicalOperator] = []
     ports: Sequence[CanonicalPort] = []
     edges: Sequence[CanonicalEdge] = []
+
+
+class CanonicalPipeline(CanonicalPipelineData):
+    id: CanonicalPipelineID
+    revision_id: int
```

Comment:
What is the distinction between these two models?

Reply:
I did this so we can use generated zod validation in frontend minus the `id` and `revision_id`, since they are not known while composing